src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/HelpersWindows.cs

```diff
@@ -225,74 +225,95 @@ public static AlgorithmIdentifier ToAlgorithmIdentifier(this CRYPT_ALGORITHM_IDE
+                                keyLength = KeyLengths.Rc2_64Bit;
+                                break;
+                            case CryptRc2Version.CRYPT_RC2_128BIT_VERSION:
+                                keyLength = KeyLengths.Rc2_128Bit;
+                                break;
+                            default:
+                                keyLength = 0;
+                                break;
+                        }
+                        // Retrieve IV if available.
+                        if (rc2Parameters.fIV > 0)
+                        {
+                            parameters = rc2Parameters.ToByteArray();
                         }
-                        break;
                     }
+                    break;
+                }
 
                 case AlgId.CALG_RC4:
+                {
+                    int saltLength = 0;
+                    if (cryptAlgorithmIdentifer.Parameters.cbData != 0)
                     {
-                        int saltLength = 0;
-                        if (cryptAlgorithmIdentifer.Parameters.cbData != 0)
+                        using (SafeHandle sh = Interop.Crypt32.CryptDecodeObjectToMemory(
+                            CryptDecodeObjectStructType.X509_OCTET_STRING, 
+                            cryptAlgorithmIdentifer.Parameters.pbData, 
+                            (int)cryptAlgorithmIdentifer.Parameters.cbData))
                         {
-                            using (SafeHandle sh = Interop.Crypt32.CryptDecodeObjectToMemory(CryptDecodeObjectStructType.X509_OCTET_STRING, cryptAlgorithmIdentifer.Parameters.pbData, (int)cryptAlgorithmIdentifer.Parameters.cbData))
+                            unsafe
                             {
-                                unsafe
-                                {
-                                    DATA_BLOB* pDataBlob = (DATA_BLOB*)(sh.DangerousGetHandle());
-                                    saltLength = (int)(pDataBlob->cbData);
-                                }
+                                DATA_BLOB* pDataBlob = (DATA_BLOB*)(sh.DangerousGetHandle());
+                                saltLength = (int)(pDataBlob->cbData);
+                                parameters = (*pDataBlob).ToByteArray();
                             }
                         }
-
-                        // For RC4, keyLength = 128 - (salt length * 8).
-                        keyLength = KeyLengths.Rc4Max_128Bit - saltLength * 8;
-                        break;
                     }
 
+                    // For RC4, keyLength = 128 - (salt length * 8).
+                    keyLength = KeyLengths.Rc4Max_128Bit - saltLength * 8;
+                    break;
+                }
+
                 case AlgId.CALG_DES:
                     // DES key length is fixed at 64 (or 56 without the parity bits).
                     keyLength = KeyLengths.Des_64Bit;
+
+                    GetParameters(cryptAlgorithmIdentifer, ref parameters);
```

Comment:
I'd expect to see a version of this code in the managed implementation (for Unix).  And I'd expect to see tests added (or Asserts added to existing tests) to confirm that the values are populated as expected.